dts/bindings/input/zephyr,lvgl-button-input.yaml

```diff
@@ -0,0 +1,47 @@
+# Copyright 2023 Fabian Blatz <fabianblatz@gmail.com>
+# SPDX-License-Identifier: Apache-2.0
+
+description: |
+  LVGL button indev pseudo-device
+
+  Listens for button input events and routes the
+  lv_indev_data_t to the underlying button lv_indev_t managed by LVGL.
+
+  Needs to be associated to a specific device to listen for events
+  from that device. Example configuration:
+
+  pointer {
+          compatible = "zephyr,lvgl-button-input";
+          input = <&buttons>;
+          input-codes = <INPUT_KEY_0>;
+          lvgl-key = <10>;
+          x-coordinate = <120>;
+          y-coordinate = <120>;
+  };
+
+compatible: "zephyr,lvgl-button-input"
```

Comment:
good stuff! you should add something that builds this in CI, the lvgl sample already takes the button to reset the counter, how about moving that code to use input as well (just let it hit `INPUT_KEY_0`, most boards have it) and then add a config on this one? May even be a separate PR if you fancy it.

Reply:
Just checked, adding  general purpose might be a bit tricky, since I at the moment the coordinates for the button are hardcoded on the device tree overlay. It's possible to assign them at runtime, but I would need to reach into the `lvgl_common_input_data` struct.  
Since the it is assumed that `lvgl_common_input_data` is the first member of any pseudo device `data` struct this would be doable, but I also adds that assumption to the sample. Otherwise a dedicated header may be added, which encapsulates the call. Something like:

```
lv_indev_t* lvgl_input_get_indev(const struct* device);
```

This will eventually be need anyways I guess, since the rotary encoders `indev` wants to be assigned a `lv_group_t` to function properly.

Reply:
Oh I was thinking about just adding an example fixed instance as part of the sample, just to have the code build, maybe pointing to an LVGL keycode instead coordinate so it does not depend on the screen size? Alternatively you could start new `tests/drivers/build_all`.

Reply:
👍 I added an `lvgl-button-input` to the `native_posix` overlay. This will trigger a click on the "hello world" button in the sample when pressing 'b' on the keyboard (I extended the sdl_gpio).